docs/upgrading.md

```diff
@@ -0,0 +1,64 @@
+
+With the increased security available with supplying TLS certificates for securing gRPC, we are recommending that users migrate to separate Contour and Envoy pods, deployed using a Deployment and Daemonset respectively. This will soon enable greater functionality for status updates on Contour-managed objects, and should help with upgrade issues eventually.
+
+The other major feature that is available is an experimental version of Leader Election. Currently, you can enable leader election with the `--enable-leader-election` flag to `contour serve`. This will ensure that *only the elected leader serves the gRPC endpoints.* That is, only the elected leader will actually open the gRPC port and start serving.
+
+There are some further requirements for this at the moment, see the howto section below for more.
+
+## The easy way to upgrade
+
+If the following are true for you:
+
+ * Your installation is in the `heptio-contour` namespace
+ * You are using one of the current example deployments
+ * Your workload can take a minute or two of downtime
+
+Then you can upgrade to 0.15 by deleting the `heptio-contour` namespace and reinstalling as follows from the root directory of the repository:
+```
+kubectl delete namespace heptio-contour
+
+kubectl apply -f examples/<your-desired-deployment>
+```
+
+## The less easy way
+
+### Required changes
+
+You must either enable TLS for gRPC serving, or put `--insecure` into your `contour serve` startup line. If you are running with both Contour and Envoy in a single pod, the existing deployment examples have already been updated with this change.
+
+If you are running using the `ds-hostnet-split` example or a derivative, we strongly recommend that you generate new certificates for securing your gRPC communication between Contour and Envoy.
+
+There is a Job in the `ds-hostnet-split` directory that will use the new `contour certgen` command to generate a CA and then sign Contour and Envoy keypairs, which can also then be saved directly to Kubernetes as Secrets, ready to be mounted into your Contour and Envoy Deployments and Daemonsets.
+
+Again, check the examples for Contour and Envoy in `ds-hostnet-split` for the exact commands.
+
+The steps you will need to follow for this to work are:
+1. Generate a CA keypair
+1. Use it to generate and sign two more keypairs, one for Contour and one for Envoy.
+1. Get the CA certificate (only) and the Contour and Envoy keypairs into Kubernetes Secrets inside the `heptio-contour` (or whatever) namespace.
+1. Mount the CA cert and the keypair into the Contour and Envoy pods, and supply the correct parameters to `contour serve` (for Contour) and `contour bootstrap` (for Envoy)
+
+If you are interested in the exact details of how the certificates need to be generated, see `docs/grpc-tls-howto.md`, which explains the `openssl` commands required to perform the same steps as `contour certgen` can do for you.
+
+
+### Optional changes
+
+#### Leader Election (EXPERIMENTAL)
```

Comment:
Does leader election belong here? It's not required for upgrading and is op-in at the moment. I just wonder if there's a better spot for it somewhere else in the docs.

Reply:
I wanted to put it in here to get some eyes on the feature. Once we've got the feature more nailed down, there should be a more dedicated document, agreed.